pandas/tests/test_frame.py

```diff
@@ -2954,6 +2954,23 @@ def test_constructor_dict_multiindex(self):
         df = df.reindex(columns=expected.columns, index=expected.index)
         check(df, expected)
 
+    def test_constructor_dict_datetime64_index(self):
+
+        data = {1: {np.datetime64('1990-03-15'): 1},
```

Comment:
for completeness, pls also do a test (you can use a loop), with ``datetime.datetime`` and ``Timestamp`` (expected should be the same)